app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,57 @@
+class MarketsController < ApplicationController
+
+  def index #VIEWED BY USER
+    @vendors = Vendor.order(name: :asc)
+    @markets = Market.order(state: :asc, city: :asc)
+    render :index
+  end
+
+  def new
+    @market = Market.new
+    render :new
+  end
+
+  def create
+    @market = Market.new(create_market_params[:market])
+    #unknown if need other things attached, like vendor/sales/products
+    if @market.save
+      @markets = Market.order(state: :asc, city: :asc)
+      redirect_to markets_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @market = Market.find(params[:id])
+    render :new
+  end
+
+  def show
+    @market = Market.find(params[:id])
+    @vendors = Vendor.where(market_id: @market.id)
+    render :show
+  end
+
+  def update
+    @market = Market.find(params[:id]).update(create_market_params[:market])
```

Comment:
Yay great job using the same create_market_params method for both create and update actions!